File: Assessment_Questions/src/test/java/com/pages/LoginPage.java
package com.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class LoginPage extends BasePage {

    public LoginPage(WebDriver driver) {
        super(driver);
    }

    @FindBy(css = "input[data-qa='login-email']")
    WebElement email;

    @FindBy(css = "input[data-qa='login-password']")
    WebElement password;

    @FindBy(css = "button[data-qa='login-button']")
    WebElement loginBtn;

    @FindBy(xpath = "//p[text()='Your email or password is incorrect!']")
    WebElement errorMsg;

    public void login(String user, String pass) {
        email.sendKeys(user);
        password.sendKeys(pass);
        loginBtn.click();
    }

    public String getErrorMsg() {
        return errorMsg.getText();
    }
}

File: Akksheetha M - Assessment 24/ECommerceAutomation/Question2/pages/login_page.py
from selenium.webdriver.common.by import By
from utils.wait import wait_for

class LoginPage:

    def __init__(self, driver):
        self.driver = driver

    def open_login(self):
        wait_for(self.driver,(By.XPATH, "//span[text()='My Account']")).click()
        wait_for(self.driver,(By.XPATH, "//ul//a[text()='Login']")).click()

    def login(self, email, password):
        wait_for(self.driver,(By.ID, "input-email")).send_keys(email)
        self.driver.find_element(By.ID, "input-password").send_keys(password)
        self.driver.find_element(By.XPATH, "//input[@value='Login']").click()
        try:
            return wait_for(self.driver,(By.XPATH, "//h2[text()='My Account']")).text
        except:
            return wait_for(self.driver,(By.XPATH, "//div[@class='alert alert-danger alert-dismissible']")).text